//==============================================================================
//                                                                            
//                       Cho sanghyun's Game Classes II
//                         Ver 9.0 / Release 2017.01.03
//                                                                            
//                        Socket Classes Tutorial (Ex8)
//                                     protobuf
//-----------------------------------------------------------------------------
//
// - protobuf message definitions
//
//
//
//
//

//-----------------------------------------------------------------------------
// Declaration
//-----------------------------------------------------------------------------
// 1) Protobuf¿« Version
syntax = "proto3";

// 2) Package∏Ì
package		tutorial;


//-----------------------------------------------------------------------------
// Definitions of 
//-----------------------------------------------------------------------------
// 1) java_declaration
option java_package			 = "com.example.tutorial";
option java_outer_classname	 = "SimpleProtos";

// 2) C# declaration
option csharp_namespace		 = "Google.Protobuf.Examples.tutorial";



//-----------------------------------------------------------------------------
// Enum
//-----------------------------------------------------------------------------
//enum eMESSAGE_COMMAND
//{
//	MESSAGE_FIRST	 = 0;
//	MESSAGE_SEND_A	 = 20;
//	MESSAGE_SEND_B	 = 21;
//	MESSAGE_SEND_C	 = 22;
//}


//-----------------------------------------------------------------------------
// Message definitions
//-----------------------------------------------------------------------------
// [Message] SendA
message SendA
{
	int32				TestValue	 = 2;
}

// [Message] SendB
message SendB
{
	int32				TestValue1	 = 2;
	uint32				TestValue2	 = 3;
	float				TestValue3	 = 4;
}

// [Message] SendC
message SendC
{
	int32				TestValue1	 = 2;
	string				TestValue2	 = 3;
	repeated int32		TestValue3	 = 4;
}
